src/WebJobs.Script/Description/PowerShell/PowerShellFunctionInvoker.cs

```diff
@@ -0,0 +1,490 @@
+                        BindingContext bindingContext = new BindingContext
+                        {
+                            Binder = binder,
+                            BindingData = bindingData,
+                            Value = stream
+                        };
+                        await inputBinding.BindAsync(bindingContext);
+                    }
+                }
+
+                environmentVariables[inputBinding.Metadata.Name] = Path.Combine(functionInstanceOutputPath, inputBinding.Metadata.Name);
+            }
+        }
+
+        private static async Task ProcessOutputBindingsAsync(string functionInstanceOutputPath, Collection<FunctionBinding> outputBindings,
+            object input, IBinderEx binder, Dictionary<string, string> bindingData)
+        {
+            if (outputBindings == null)
+            {
+                return;
+            }
+
+            try
+            {
+                foreach (var outputBinding in outputBindings)
+                {
+                    string filePath = System.IO.Path.Combine(functionInstanceOutputPath, outputBinding.Metadata.Name);
+                    if (File.Exists(filePath))
+                    {
+                        using (FileStream stream = File.OpenRead(filePath))
+                        {
+                            BindingContext bindingContext = new BindingContext
+                            {
+                                Input = input,
+                                Binder = binder,
+                                BindingData = bindingData,
+                                Value = stream
+                            };
+                            await outputBinding.BindAsync(bindingContext);
+                        }
+                    }
+                }
+            }
+            finally
+            {
+                // clean up the output directory
+                if (outputBindings.Any() && Directory.Exists(functionInstanceOutputPath))
+                {
+                    Directory.Delete(functionInstanceOutputPath, recursive: true);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Event handler for the output stream.
+        /// </summary>
+        private void OutputCollectionDataAdded(object sender, DataAddedEventArgs e)
+        {
+            // do something when an object is written to the output stream
+            var source = (PSDataCollection<PSObject>)sender;
+            var msg = source[e.Index].ToString();
+            TraceWriter.Info(msg);
+        }
+
+        /// <summary>
+        /// Event handler for the error stream.
+        /// </summary>
+        private void ErrorDataAdded(object sender, DataAddedEventArgs e)
+        {
+            var source = (PSDataCollection<ErrorRecord>)sender;
+            var msg = GetErrorMessage(source[e.Index]);
+            TraceWriter.Error(msg);
+        }
+
+        private string GetErrorMessage(ErrorRecord errorRecord)
```

Comment:
Need to add lots of unit tests. Methods like this and some of your other methods are easily unit testable.